### DISCUSSION TITLE
Building a new plug-in

### DISCUSSION THREAD

**Discussion by neuropacman** [nroduit/Weasis#605] [Q&A]
Hi All,

I am attempting to build a new plug-in for the Weasis system. I have noticed that the documentation seems to be out-of-date, hence I've been having a hard time getting things to work properly.

I am simply trying to achieve the first step of getting the plug-in to appear in the interface. Following the docs, I have generate a plug-in template using the "weasis-plugin-dicom-viewer-archetype". Without modifying any code besides naming, I have been attempting to add the plug-in to the interface. I see in the docs that "This documentation has not been updated since version 4.2.0 where the properties configuration files have been replaced by json files."

Within the Weasis configuration files, I have found a base.json file that seems to be similar in content to the ext-config.properties file used by the provided "weasis-isowriter" example (ex. "felix.auto.start.85=${weasis.codebase.url}/plugins/weasis-isowriter-2.6.1.jar"). I have also modified the pom.xml file to reflect the correct value for the relativePath property as per the documentation. 

Additionally, I have included a JAR file of my plug-in in the "/weasis/plugins" folder. Although I could not locate this directory, so I created it in a few different places to ensure that Weasis could locate it. I had no luck so I had to actually install Weasis rather than running it locally so that "/Applications/Weasis.app/Contents/app/plugins" was created by the application and I put my JAR file there.

After doing so, I see in boot.log a message "23.10.2024 19:36:53.371 *INFO * Uninstall SNAPSHOT bundle: neuropacs-weasis-plugin-1.0-SNAPSHOT", seemingly indicating that Weasis has located my plug-in but something is wrong with it's configuration.

I am also new to OSGi, so I'm sure I am missing something. Any help is greatly appreciated!

I can also provide my pom.xml file if necessary.

**Comment by nroduit** [comment]
The [documentation](https://weasis.org/en/basics/customize/build-plugins/) has been updated.
